### PR TITLE
Update lite interpreter tutorial to beta

### PR DESCRIPTION
**Pull request by cccclai**
Update lite interpreter tutorial (from ptototype to beta) with new instructions

### CODE REVIEW

File: prototype_source/lite_interpreter.rst [hunk 1]
```diff
@@ -28,51 +32,31 @@ Get ImageSegmentation demo app in Android: https://github.com/pytorch/android-de
...
-        repositories {
-            google()
-            jcenter()
-            flatDir {
-                dirs 'libs'
-            }
-        }
-    }
 
 4. **Update model loader api**: Update ``ImageSegmentation/app/src/main/java/org/pytorch/imagesegmentation/MainActivity.java`` by
```
Comment (jeffxtang) [inline prototype_source/lite_interpreter.rst:61]: No more step 3?

Comment (cccclai) [reply prototype_source/lite_interpreter.rst:61]: After 1.9, no more step 3

Comment (cccclai) [reply prototype_source/lite_interpreter.rst:61]: Good catch, it should be step 3 now

Comment (cccclai) [reply prototype_source/lite_interpreter.rst]: Fix in latest commit. Also update the name `pytorch_android-release.aar`

File: prototype_source/lite_interpreter.rst [hunk 2]
```diff
@@ -28,51 +32,31 @@ Get ImageSegmentation demo app in Android: https://github.com/pytorch/android-de
...
-
-Update `all projects` part in ``ImageSegmentation/build.gradle`` to
+    repositories {
+        maven {
+            url "https://oss.sonatype.org/content/repositories/snapshots"
+        }
+    }
 
+    dependencies {
+        implementation fileTree(dir: "libs", include: ["*.jar"])
```
Comment (jeffxtang) [inline prototype_source/lite_interpreter.rst:46]: not needed anymore

File: prototype_source/lite_interpreter.rst [hunk 3]
```diff
@@ -28,51 +32,31 @@ Get ImageSegmentation demo app in Android: https://github.com/pytorch/android-de
...
+        implementation 'androidx.appcompat:appcompat:1.2.0'
+        implementation 'androidx.constraintlayout:constraintlayout:2.0.2'
+        testImplementation 'junit:junit:4.12'
+        androidTestImplementation 'androidx.test.ext:junit:1.1.2'
+        androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
+        implementation 'org.pytorch:pytorch_android:1.9.0-SNAPSHOT'
+        implementation 'org.pytorch:pytorch_android_torchvision:1.9.0-SNAPSHOT'
+
+        implementation 'com.android.support:appcompat-v7:28.0.0'
```
Comment (jeffxtang) [inline prototype_source/lite_interpreter.rst:55]: not needed anymore - this line and next line.

File: prototype_source/lite_interpreter.rst [hunk 4]
```diff
@@ -28,53 +32,33 @@ Get ImageSegmentation demo app in Android: https://github.com/pytorch/android-de
     # Export lite interpreter version model (compatible with lite interpreter)
     scripted_module._save_for_lite_interpreter("deeplabv3_scripted.ptl")
```
Comment (jeffxtang) [inline prototype_source/lite_interpreter.rst:33]: The `optimize_for_mobile` call is strongly recommended:
```
from torch.utils.mobile_optimizer import optimize_for_mobile
optimized_scripted_module = optimize_for_mobile(scripted_module)
optimized_scripted_module._save_for_lite_interpreter("deeplabv3_scripted_optimized.ptl")
```

In my test, using `deeplabv3_scripted_optimized.ptl`, the inference time speeds up about 100%, from over 3s (for deeplabv3_scripted.ptl` to about 1.5s.

Comment (cccclai) [reply prototype_source/lite_interpreter.rst:33]: Oh good catch. Will update it.

Comment (cccclai) [reply prototype_source/lite_interpreter.rst]: Update in the last commit

File: prototype_source/lite_interpreter.rst [hunk 5]
```diff
@@ -85,38 +68,29 @@ Update `all projects` part in ``ImageSegmentation/build.gradle`` to
...
-In the build settings, search for **other linker flags**. Add a custom linker flag below
-```
--all_load
-```
-Finally, disable bitcode for your target by selecting the Build Settings, searching for Enable Bitcode, and set the value to **No**.
-
-5. **Update library and api**
-
-  5.1 Update ``TorchModule.mm``: To use the custom built libraries the project, replace `#import <LibTorch/LibTorch.h>` (in ``TorchModule.mm``) which is needed when using LibTorch via Cocoapods with the code below:
+  3.1 Update ``TorchModule.mm``: To use the custom built libraries the project, replace `#import <LibTorch/LibTorch.h>` (in ``TorchModule.mm``) which is needed when using LibTorch via Cocoapods with the code below:
```
Comment (raziel) [inline prototype_source/lite_interpreter.rst:93]:
```suggestion
  3.1 Update ``TorchModule.mm``: To use the custom built libraries project, replace `#import <LibTorch/LibTorch.h>` (in ``TorchModule.mm``) which is needed when using LibTorch via CocoaPods with the code below:
```

File: prototype_source/lite_interpreter.rst [hunk 6]
```diff
@@ -175,10 +148,12 @@ Finally, disable bitcode for your target by selecting the Build Settings, search
         fatalError("Can't find the model file!")
     }
 
-6. Build and test the app in Xcode.
+4. Build and test the app in Xcode.
 
-How to use lite interpreter + custom build
+How to use mobile interpreter + custom build
 ------------------------------------------
+Custom PyTorch library only contains the operators needed by the model, to do that:
```
Comment (raziel) [inline prototype_source/lite_interpreter.rst:155]:
```suggestion
A custom PyTorch interpreter library can be created to reduce binary size, by only containing the operators needed by the model. In order to do that follow these steps:
```

File: prototype_source/lite_interpreter.rst [hunk 7]
```diff
@@ -198,19 +173,20 @@ In the snippet above, you first need to load the ScriptModule. Then, use export_
...
-   SELECTED_OP_LIST=deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86
+   SELECTED_OP_LIST=deeplabv3_scripted.yaml ./scripts/build_pytorch_android.sh x86
+
 
 
 Conclusion
 ----------
 
-In this tutorial, we demonstrated how to use lite interpreter in Android and iOS app. We walked through an Image Segmentation example to show how to dump the model, build torch library from source and use the new api to run model. Please be aware of that lite interpreter is still under development, more library size reduction will be introduced in the future. APIs are subject to change in the future versions.
+In this tutorial, we demonstrated how to use mobile interpreter in Android and iOS app. We walked through an Image Segmentation example to show how to dump the model, build torch library from source and use the new api to run model. Please be aware of that mobile interpreter is still under development, more library size reduction will be introduced in the future. APIs are subject to change in the future versions.
```
Comment (raziel) [inline prototype_source/lite_interpreter.rst:189]:
```suggestion
In this tutorial, we demonstrated how to use PyTorch's efficient mobile interpreter, in an Android and iOS app.

We walked through an Image Segmentation example to show how to dump the model, build a custom torch library from source and use the new api to run model.

Our efficient mobile interpreter is still under development, and we will continue improving its size in the future. Note, however, that the APIs are subject to change in future versions.
```

File: prototype_source/lite_interpreter.rst [hunk 8]
```diff
@@ -198,19 +173,20 @@ In the snippet above, you first need to load the ScriptModule. Then, use export_
...
 
 Thanks for reading! As always, we welcome any feedback, so please create an issue `here <https://github.com/pytorch/pytorch/issues>`_ if you have any.
```
Comment (raziel) [inline prototype_source/lite_interpreter.rst:191]:
```suggestion
Thanks for reading! As always, we welcome any feedback, so please create an issue `here <https://github.com/pytorch/pytorch/issues>` - if you have any.
```

**Review by raziel — COMMENTED** [review]
Thanks @cccclai 

Please take a look.

**Review by raziel — COMMENTED** [review]
Thanks for taking the suggestions.

**Pull request merged**